src/NUnitFramework/framework/Constraints/NUnitEqualityComparer.cs

```diff
@@ -154,7 +154,24 @@ public IList<FailurePoint> FailurePoints
         /// <summary>
         /// Compares two objects for equality within a tolerance.
         /// </summary>
-        public bool AreEqual(object x, object y, ref Tolerance tolerance, bool topLevelComparison = true)
+        public bool AreEqual(object x, object y, ref Tolerance tolerance)
+        {
+            return AreEqual(x, y, ref tolerance, true);
+        }
+
+        /// <summary>
+        /// Compares two objects for equality within a tolerance.
+        /// </summary>
+        public bool AreEqual(object x, object y, ref Tolerance tolerance, bool topLevelComparison)
+        {
+            return AreEqual(x, y, ref tolerance, new ComparisonState()
+            {
+                TopLevelComparison = topLevelComparison,
+                Comparisons = new List<ComparisonState.Comparison>()
+            });
```

Comment:
A constructor should initialize all required fields. It wouldn't make sense for `Comparisons` to be null, so I'd suggest a `bool` constructor that creates the list.